nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,2575 @@
+            },
+            "text/plain": [
+              "  0%|          | 0/3 [00:00<?, ?it/s]"
+            ]
+          },
+          "metadata": {}
+        },
+        {
+          "output_type": "stream",
+          "name": "stderr",
+          "text": [
+            "Loading cached processed dataset at /root/.cache/huggingface/datasets/emotion/default/0.0.0/348f63ca8e27b3713b6c04d723efe6d824a56fb3d1449794716c0f0296072705/cache-7161aca97360dca6.arrow\n",
+            "Loading cached processed dataset at /root/.cache/huggingface/datasets/emotion/default/0.0.0/348f63ca8e27b3713b6c04d723efe6d824a56fb3d1449794716c0f0296072705/cache-51b5751515a95f0f.arrow\n",
+            "Loading cached processed dataset at /root/.cache/huggingface/datasets/emotion/default/0.0.0/348f63ca8e27b3713b6c04d723efe6d824a56fb3d1449794716c0f0296072705/cache-3ad9ee5140e9d78b.arrow\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RQ24-mWS6elF"
+      },
+      "source": [
+        "The dataset is already split into 16,000 train, 2,000 validation and 2,000 test samples. To investigate the effectiveness of the GPT3Mix augmentation method, we will use only 10 samples per class as a train set."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "9YI6-pZdSqnI",
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "outputId": "60d03419-bc4b-4f5c-ca74-3c4ebe633043"
+      },
+      "source": [
+        "# select 10 random train samples from each of the three emotions\n",
+        "# sadness (0), joy (1), love (2), anger (3), fear (4), surprise (5)\n",
+        "joy_train_samples = emotion_ds['train'].filter(lambda e: e['label'] == 1).select(range(10))\n",
+        "anger_train_samples = emotion_ds['train'].filter(lambda e: e['label'] == 3).select(range(10))\n",
+        "surprise_train_samples = emotion_ds['train'].filter(lambda e: e['label'] == 5).select(range(10))\n",
+        "\n",
+        "# map emotions to integers for labeling\n",
+        "# joy (0), anger (1), surprise (2)\n",
+        "def map_emotions(example):\n",
+        "  example['label'] = example['label']//2\n",
+        "  return example\n",
+        "\n",
+        "# create a train set that consists of 10 samples per class and filter the test \n",
+        "# set to contain only the valid labels\n",
+        "emotion_train_ds = concatenate_datasets([joy_train_samples, anger_train_samples, surprise_train_samples]).map(lambda e: map_emotions(e)).shuffle(seed=42)\n",
+        "emotion_test_ds = emotion_ds[\"test\"].filter(lambda e: e['label'] in [1, 3, 5]).map(lambda e: map_emotions(e))\n",
+        "\n",
+        "# define the maping between emotions and labels\n",
+        "idx2label = {0: 'joy', 1: 'anger', 2: 'surprise'}\n",
+        "label2idx = {'joy': 0, 'anger': 1, 'surprise': 2}"
+      ],
+      "execution_count": null,
+      "outputs": [
```

Comment:
In general, I would go over the outputs here and there and see where it makes sense to keep them and where to clear them